graph-deploy.js

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env node
+
+let app = require('commander')
+let request = require('request')
+let args = require('./src/cli/args')
+const url = require('url')
+
+args.addBuildCommand()
+
+app
+    .option(
+        '-n, --subgraph-name <NAME>',
+        'subgraph name'
+    )
+    .option(
+        '--node <URL>[:PORT]',
+        'graph node url'
+    )
+
+app.parse(process.argv)
+
+if (!app.node || !app.subgraphName) {
+    app.help()
+}
+
+let compiler = args.compilerFromArgs()
+
+let requestUrl = new URL(app.node);
+if (!requestUrl.port) {
+    requestUrl.port = "8020"
+}
+
+compiler.compile()
+    .then(
+        function (ipfsHash) {
+            if (ipfsHash === undefined) {
+                console.log("compilation failed, not deploying")
+                return
+            }
+            requestUrl.pathname += `subgraph_deploy/${app.name}/${ipfsHash}`
+            request.post(requestUrl, {
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            },
+                function (error, res, body) {
+                    if (error) {
+                        console.log("error sending json-rpc request " + error)
```

Comment:
Again: `console.error`. Also, since this a user facing error, I think formatting is important. I'd do:
```
console.error('Failed to deploy the subgraph:', error)
```

Reply:
Actually, we probably want to use a logger like in the compiler and type generator for this, to get consistent colors etc.